Swent-team-6/icebreakrr
# end to end

Here is the PR including the end to end tests for Milestone. There are 4 end to end files in total : 
- `AroundYouFlowEndToEnd.kt` which contains the user flow of the around you screen except the AI part : 
     - click on a profile
     - test the behavior of the dialogs to send requests without sending it
     - filter screen and check  it's influence on the around you screen
- `SettingsFlowEndToEnd.kt` contains the user flow of the setting screen and all of it's features except the deletion of the account : 
     - check if our profile is well displayed
     - modify our profile and check that it has been updated
     - test the block functionality
     - test the already met functionality
     - test the toggle discoverability feature
     - test the report
     - test the log out
- `ReceiveRequestFlowEndToEnd.kt` contains multiple end to end tests that check the behavior of the meeting request system when you receive a request : 
     - receive a request and then a cancellation message TIME, CLOSED, DISTANCE or CANCELLED and check everything is updated
     - receives a request and decline it
     - receive a request and accept it
- `SendRequestFlowEndToEnd.kt` contains multiple end to end tests that check the behavior of the meeting request system when you send a request except the time cancellation because this feature is disabled for the moment 
     - send a request and then receive a cancellation message  CLOSED, DISTANCE or CANCELLED and check everything is updated
     - send a request and then cancel your request
     - send a request and receive a positive response
     - send a request and receive a negative response


I also renamed `MainActivity` to `Icebreakrr`

Here is the jacoco test report, we have 82% coverage on the project overall : 

![image](https://github.com/user-attachments/assets/a6e1355f-bc2d-4dda-800e-8652fcfb8aaa)
